internal/k8s/client.go

```diff
@@ -0,0 +1,166 @@
+package k8s
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/Shopify/kubeaudit/k8stypes"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// ErrNoReadableKubeConfig represents any error that prevents the client from opening a kubeconfig file.
+var ErrNoReadableKubeConfig = errors.New("unable to open kubeconfig file")
+
+var DefaultClient = K8sClient{}
+
+// Client abstracts the API to allow testing.
+type Client interface {
+	InClusterConfig() (*rest.Config, error)
+}
+
+// K8sClient wraps kubernetes client-go so it can be mocked.
+type K8sClient struct{}
+
+// InClusterConfig wraps the client-go method with the same name.
+func (kc K8sClient) InClusterConfig() (*rest.Config, error) {
+	return rest.InClusterConfig()
+}
+
+func NewKubeClientLocal(configPath string) (*kubernetes.Clientset, error) {
+	if _, err := os.Stat(configPath); err != nil {
+		return nil, ErrNoReadableKubeConfig
+	}
+	config, err := clientcmd.BuildConfigFromFlags("", configPath)
+	if err != nil {
+		return nil, err
+	}
+	kube, err := kubernetes.NewForConfig(config)
```

Comment:
Is there a reason we're not using the KUBECONFIG environment variable to determine the configuration to use? We do this here: https://github.com/Shopify/vouch4cluster/blob/master/cmd/kube.go#L31

Reply:
I think kubeaudit just never supported using that environment variable. From the readme:

> Local config mode: If kubeaudit is not running in a container, `kubeaudit cmd` will audit the resources specified by your local kubeconfig (`$HOME/.kube/config`) file. You can force kubeaudit to use a specific local config file with the switch `-c/--kubeconfig /config/path`

ref: https://github.com/Shopify/kubeaudit#general-instructions

I'm not against changing it, but do you think it should be part of this PR?

Reply:
I think this shouldn't be done in this PR, this PR is already overwhelming as it is :) let's create an issue and come back to it in a different PR

Reply:
Sorry, I thought I responded! I didn't realize they were different structures, ignore me! I'm not even certain this warrants an issue, I thought it was a quick optimization but looking at those structures it does not look it.

Reply:
https://github.com/Shopify/kubeaudit/issues/242